src/dataflow/src/sink/kafka.rs

```diff
@@ -176,105 +183,122 @@ pub fn kafka<G>(
         &name.clone(),
         |info| {
             let activator = stream.scope().activator_for(&info.address[..]);
-            move |input| {
-                // Grab all of the available Rows and put them in a queue before we
-                // send it over to Kafka. We Even though we want to do bounded work
-                // per sink invocation, we still need to remember all inputs as we
-                // receive them
-                input.for_each(|_, rows| {
-                    rows.swap(&mut vector);
+            let shutdown_button_ref = &mut shutdown_button;
+            *shutdown_button_ref = Some(ShutdownButton::new(
+                producer.clone(),
+                stream.scope().activator_for(&info.address[..]),
+            ));
```

Comment:
I'm not sure I understand what this is all about. Why can't you ditch the option like this?

```rust
            let shutdown_button = ShutdownButton::new(
                producer.clone(),
                stream.scope().activator_for(&info.address[..]),
            );
```

Reply:
I was copying the pattern that was used for creating shutdown buttons in differential but agreed it doesnt seem necessary in this case so I've removed. thanks for the catch!